### DISCUSSION TITLE
Help me to change https://localhost:3000 to https://local-domain.com 🙏

### DISCUSSION THREAD

**Discussion by mocsimas** [nuxt/nuxt#23875] [Questions] [answered]
I'm having a 3 days old problem with no luck.

I'm developing an SPA with Nuxt 3 on the frontend and Laravel as API on the backend. My objective is to modify the URL of the server provided by `npm run dev` command, changing it from https://localhost:3000 to https://local-domain.com.

I came to this idea after social oauth providers required a callback URL with valid domain. In this case localhost:3000 does not meet the requirements.
My backend API is running at virtual host https://api.local-domain.com. API authentication is served by Laravel Sanctum and Laravel Socialite. 

I managed to create a virtual host proxying from https://localhost:3000 to https://local-domain.com. At first, it seemed to work, but I received CSRF token mismatch errors on all requests and whenever I tried to make the same requests with https://localhost:3000 - the errors disappeared.

This is my `nuxt.config.ts` file contents:
```typescript
import { fileURLToPath } from 'url'

// https://nuxt.com/docs/api/configuration/nuxt-config
export default defineNuxtConfig({
  devtools: { enabled: true },

  runtimeConfig: {
    public: {
      api_url: import.meta.env.API_URL,
    }
  },

  modules: [
    ['@pinia/nuxt'],
    ['@nuxtjs/tailwindcss'],
  ],

  imports: {
    autoImport: false
  },

  plugins: [
    '~/plugins/loadUser.ts'
  ],

  css: [
    '~/assets/scss/app.scss',
  ],

  devServer: {
    https: {
      key: './server.key',
      cert: './server.crt',
    },
  },

  vite: {
    server: {
      hmr: {
        clientPort: 3000
      },
    }
  },
})

``` 

What can be done or are the any alternatives I could do to solve this problem? Any help is highly appreciated 🙏

**Comment by OnlyWick** [comment, marked as answer]
Did you try:

```
devServer: {
    host: 'local-domain.com',
    // ...
  },
```

In addition, you need to config `hosts`(`/etc/hosts` in macOS) in you computer. If you have any proxy tool in you computer, you need to check the filter of proxy tool(or exit proxy tool).

**Reply by mocsimas** [reply]
Did the trick ;) thanks!

**Comment by lukas-pierce** [comment]
Fixed here

https://github.com/crxjs/chrome-extension-tools/issues/696#issuecomment-1868897955
